### PR TITLE
Passiv Solar Data Format

### PR DESCRIPTION
**Pull request by ldodds**
The passiv system for collecting solar data is only able to produce CSV files that contain a timestamp for the day the readings were _collected_ rather than when the energy was consumed. So the timestamps are always a day behind.

As they are unable to change this, this PR:

* Adds a new option to our data feed configs to flag that there are "delayed readings"
* Refactors the existing data feed translator spec to clarify what is currently being tested and just tidy up. The test names didn't really reflect what was being asserts in many cases
* Changes the data feed translator to reformat the dates to a day earlier so we end up with the correct dates in the database
* Adds a configuration for the Passiv format, which uses this setting
* Adds a warning on the upload page for admins about the odd date behaviour

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -1013,17 +1014,13 @@
     t.datetime "finished_at", precision: nil
     t.text "error"
     t.integer "error_event", limit: 2
```
Comment (tbhi) [inline db/schema.rb:1016]: not sure about these?

**Review by tbhi — APPROVED** [review]

File: bin/importmap [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env ruby
 
-require_relative "../config/application"
-require "importmap/commands"
+require_relative '../config/application'
```
[the remark this answers is not in the document]
Comment (ldodds) [reply bin/importmap:3]: This was an accidental commit, not sure why it was changed.

File: db/schema.rb (hunk 1 above)
Comment (ldodds) [reply db/schema.rb:1016]: I don't actually know what these changes are, all I've done is migrated. I've migrated and rolled back a few times, might try rebuilding my database.

**Review by tbhi — APPROVED** [review]

File: app/services/amr/data_feed_translator.rb [hunk 3]
```diff
@@ -45,6 +45,24 @@ def meter_details_from_row(row)
...
+      return date_string unless @config.delayed_reading
+
+      # a delayed reading config means the date/date-time column is when the readings
+      # where collected, rather than the date the energy was consumed. For now
+      # this only appears in one config where the readings are collected a day later
+      begin
+        date = DateTime.strptime(date_string, @config.date_format)
+        date = date - 1.day
+        date.strftime(@config.date_format)
+      rescue ArgumentError
```
[the remark this answers is not in the document]
Comment (ldodds) [reply app/services/amr/data_feed_translator.rb:59]: That would require further changes to check for Date:Error in later steps to reject invalid results.

**Pull request merged**